crates/ide_completion/src/render/function.rs

```diff
@@ -55,8 +56,34 @@ impl<'a> FunctionRender<'a> {
     }
 
     fn detail(&self) -> String {
```

Comment:
Hm, so I am not exactly sold on this. The problem is that in some editors, the popup is reeealy tiny, and showing parmeters obscures the result type, which is somewhat more important. 


I like how it is solved in IDEA: 

![image](https://user-images.githubusercontent.com/1711539/110351841-49d9ab00-8046-11eb-87f9-abc510eb230e.png)


There are just separate slots for params and return type. I think we should internally move towards that model, and then allow the editor to customize things.


One way we could go is splitting `detail` into `left-detail` and `right-detail`. Another is adding explicit `type` and `params` accessors. The latter seems preferable to me. 

So, let's go the long-refactory way here. 

* add (private) `enum Detail { Simple(String), Structured { params: Option<String>, ty: Option<String> } }` next to `CompletionItem`
* change the type of `detail` to `Option<Detail>`
* add accesors: `detail_params(&self) -> Option<&str>, detail_ty(&self) -> Option<&str>` (leaving the `detail` in)
* add `completion.showFunctionParametrsInPopup: true`
* in the conv.rs (?, maybe handlers) use that to either render the full thing or only the type.

Reply:
Hm, now that I've noticed #7565, I am  having doubts about the proposed API. I think what we really want to do is to fold `label` and `detail` into a single thing maybe? (five minutes walking around the room later) ... or maybe not? 

 So there's definitely a concept of `CompletionItemDisplay` --  the stuff that is *shown* in the completion popup. On the other hand, having a separate type for that doesn't seem useful: instead of `completion_item.display().foo()` we can always say just `completion_item.foo()`.

Let's maybe keep this simple and just add `params` and `ty` fields alongside `detail`. I bet most users of `detail` can migrate to `ty`, and then we'll see what's left

Reply:
I pushed a change in this direction, but I'm not sure if I properly understood what you were looking for. 

1.  I used `params_display` and `ty_display`, because `params` already existed.
2. `detail` doesn't want to print `-> ()` for unit return types, although users of `ty_display` probably would, so `detail` still has to have a conditional on the return type.